File: News_Service_Jooq/src/main/java/com/dunice/projectNews/dto/response/PageableResponse.java
package com.dunice.projectNews.dto.response;

import lombok.AllArgsConstructor;
import lombok.EqualsAndHashCode;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@EqualsAndHashCode
public class PageableResponse<T> {

    private T content;

    private Integer numberOfElements;

    public PageableResponse(T content) {

        this.content = content;
    }
}

File: News_Service_Kotlin/src/main/kotlin/com/dunice/advancedServiceKotlin/dto/response/PageableResponse.kt
package com.dunice.advancedServiceKotlin.dto.response

import java.util.*

class PageableResponse<T> {
    private var content: T? = null

    @JvmField
    var numberOfElements: Int? = null

    constructor(content: T) {
        this.content = content
    }

    fun getContent(): T? {
        return content
    }

    fun setContent(content: T) {
        this.content = content
    }

    constructor(content: T, numberOfElements: Int?) {
        this.content = content
        this.numberOfElements = numberOfElements
    }

    constructor()


    override fun equals(o: Any?): Boolean {
        if (this === o) return true
        if (o == null || javaClass != o.javaClass) return false
        val that = o as PageableResponse<*>
        return content == that.content && numberOfElements == that.numberOfElements
    }

    override fun hashCode(): Int {
        return Objects.hash(content, numberOfElements)
    }
}
